org.aspectj.matcher/src/main/java/org/aspectj/weaver/UnresolvedType.java

```diff
@@ -161,6 +161,10 @@ public final boolean isArray() {
 		return signature.length() > 0 && signature.charAt(0) == '[';
 	}
 
+	public final int getDimensions() {
+		return signature.replaceAll("^(\\[*).*", "$1").length();
```

Comment:
@aclement, do you think it would be better to eagerly (constructor) or lazily (first `getDimensions()` call) calculate this value once, assign it to a member and then just fetch it when needed, instead of doing the regex search & replace multiple times? It might not affect performance much, so I am not sure if you would rather spend CPU cycles or memory for an `int` instance field.

Reply:
My immediate reaction is that this would be overridden by resolved types and implemented more intelligently there, so it'll only be impacting folks asking getDimensions on unresolved types. It would increase the size of all unresolved types (and there could be quite a few of those at certains stages of weaving). Like you I'm skeptical of fixing if it isn't a known problem.